java/drivers/driver-hazelcast4plus/src/main/java/com/hazelcast/simulator/tests/vector/VectorCollectionPutDatasetTest.java

```diff
@@ -1,155 +1,97 @@
+                break;
+            }
+            var vector = reader.getTrainVector(key % testDataSetSize);
+            buffer.put(key, VectorDocument.of(key, VectorValues.of(vector)));
+        }
 
-    public static class ThreadState extends BaseThreadState {
+        collection.putAllAsync(buffer)
+                .toCompletableFuture()
+                .join();
     }
 
+    /**
+     * Deletes entries 0..requested size
+     */
+    @TimeStep(prob = 0)
+    public void delete() {
+        var size = getRequestedSize();
 
-    public static class TimeMetrics {
-        private static final Histogram bufferTimer = new Histogram(2);
-        private static final Histogram putTimer = new Histogram(2);
-
-
-        public void recordBuffer(Runnable action) {
-            var start = System.currentTimeMillis();
-            action.run();
-            bufferTimer.recordValue(System.currentTimeMillis() - start);
-        }
-
-        public void recordPut(Runnable action) {
-            var start = System.currentTimeMillis();
-            action.run();
-            putTimer.recordValue(System.currentTimeMillis() - start);
+        var index = counter.getAndIncrement();
+        if (index >= size) {
+            testContext.stop();
+            return;
         }
+        collection.deleteAsync(index)
+                .toCompletableFuture()
+                .join();
+    }
 
-        public String getStatistics() {
-            return "\nBuffer 95p: " + bufferTimer.getValueAtPercentile(95) + "\n"
-                    + "Buffer max: " + bufferTimer.getMaxValue() + "\n"
-                    + "Buffer min: " + bufferTimer.getMinValue() + "\n"
-                    + "Buffer mean: " + bufferTimer.getMean() + "\n"
-                    + "Put 95p: " + putTimer.getValueAtPercentile(95) + "\n"
-                    + "Put max: " + putTimer.getMaxValue() + "\n"
-                    + "Put min: " + putTimer.getMinValue() + "\n"
-                    + "Put mean: " + putTimer.getMean() + "\n";
+    @TimeStep(prob = 0)
+    public void optimize() throws ExecutionException, InterruptedException {
+        var start = System.currentTimeMillis();
+
+        // TODO: this may fail with > 1 client
+        var cleanupTimer = withTimer(() -> collection.optimizeAsync().toCompletableFuture().join());
+
+        if (backupCount + asyncBackupCount > 0) {
+            // optimize backups are async - wait for them also to get true time
+            HazelcastUtils.waitForClusterSafeState(targetInstance);
+            var cleanupTotalTimer = System.currentTimeMillis() - start;
+            logger.info("Cleanup primary time: {} ms", cleanupTimer);
+            logger.info("Cleanup wait for backups time: {} ms", cleanupTotalTimer -cleanupTimer);
+            logger.info("Cleanup total time: {} ms", cleanupTotalTimer);
+        } else {
+            logger.info("Cleanup time: {} ms", cleanupTimer);
         }
+        testContext.stop();
```

Comment:
As far as I remember, this will cause the entire test to stop, meaning the test will finish after running a single optimization. Is that what you expect?”